OpenRA.Mods.Common/Activities/UnloadCargo.cs

```diff
@@ -56,40 +59,44 @@ IEnumerable<CPos> BlockedExitCells(Actor passenger)
 		public override Activity Tick(Actor self)
 		{
 			cargo.Unloading = false;
-			if (IsCanceled || cargo.IsEmpty(self))
+			if (IsCanceled || !actorsToUnload.Any())
 				return NextActivity;
 
 			if (cloak != null && cloak.Info.UncloakOnUnload)
 				cloak.Uncloak();
 
-			var actor = cargo.Peek(self);
-			var spawn = self.CenterPosition;
-
-			var exitSubCell = ChooseExitSubCell(actor);
-			if (exitSubCell == null)
+			foreach (var actor in actorsToUnload.ToArray())
```

Comment:
A much simpler solution that wouls save you the collection transgormations, and then removals, would be to just count how many actors you have unloaded and at the end check if that matches actorsToUnload.Count